frontend/src/components/readsets/ReadsetsListContent.tsx

```diff
@@ -1,105 +1,361 @@
+        const allReadsetsReleasedOrBlocked = readsetStates.every((readsetState) => getCurrentReleaseStatus(readsetState) !== ReleaseStatus.AVAILABLE)
+        const someReadsetsChangedStatus = readsetStates.some((readsetState) => readsetState.new !== undefined)
+
+        const releaseAllEnabled =             canUpdateReleaseStatus && !allReadsetsReleased      
+        const blockAllEnabled =               canUpdateReleaseStatus && !allReadsetsBlocked       
+        const undoAllReleaseAndBlockEnabled = canUpdateReleaseStatus && !allReadsetsAvailable && isAdmin
+        const undoChangesEnabled =            canUpdateReleaseStatus && someReadsetsChangedStatus 
+        const saveChangesEnabled =            canUpdateReleaseStatus && someReadsetsChangedStatus && (
+            // normal user
+            (!isAdmin && allReadsetsReleasedOrBlocked)
+            ||
+            // admin
+            (isAdmin && (allReadsetsReleasedOrBlocked || allReadsetsAvailable))
+        )
 
-                return newState
+        let newReleaseCount = 0
+        let newBlockCount = 0
+        let newAvailableCount = 0
+        for (const readsetStatus of readsetStates) {
+            switch (readsetStatus.new) {
+                case ReleaseStatus.RELEASED:
+                    newReleaseCount++
+                    break
+                case ReleaseStatus.BLOCKED:
+                    newBlockCount++
+                    break
+                case ReleaseStatus.AVAILABLE:
+                    newAvailableCount++
+                    break
             }
         }
-    }
+        const popconfirmTitle = [
+            `From the selected dataset readsets:`,
+            newReleaseCount > 0 ? `${newReleaseCount} will be released` : '',
+            newBlockCount > 0 ? `${newBlockCount} will be blocked` : '',
+            newAvailableCount > 0 ? `${newAvailableCount} will be made available` : '',
+            `Do you want to save these changes?`
+        ].map((str, index) => <div key={index}>{str}</div>)
+    
+        return <div>
+            <Button
+                style={{ margin: 6 }}
+                onClick={() => {
+                    releaseStatusManager.setAllReleaseStatus(ReleaseStatus.RELEASED)
+                }}
+                disabled={!releaseAllEnabled}>
+                Release All
+            </Button>
+            <Button
+                style={{ margin: 6 }}
+                onClick={() => {
+                    releaseStatusManager.setAllReleaseStatus(ReleaseStatus.BLOCKED)
+                }}
+                disabled={!blockAllEnabled}>
+                Block All
+            </Button>
+            {isAdmin && <Button
+                    style={{ margin: 6 }}
+                    onClick={() => {
+                        releaseStatusManager.setAllReleaseStatus(ReleaseStatus.AVAILABLE)
+                    }}
+                    disabled={!undoAllReleaseAndBlockEnabled}>
+                    Undo All Release/Block
```

Comment:
This is confusingly similar to the other undo button. I would rename this to Reset Released Status and move it to the right of the save change button (maybe even use a different colour scheme). Also it might be nice to have it disabled unless there is some released_status that are different from Available ...